pkg/controller/perconaservermongodb/custom_users.go

```diff
@@ -354,11 +406,80 @@ func createUser(
 		secret.Annotations = make(map[string]string)
 	}
 
-	secret.Annotations[annotationKey] = string(sha256Hash(secret.Data[user.PasswordSecretRef.Key]))
+	secret.Annotations[annotationKey] = string(sha256Hash(secret.Data[passKey]))
 	if err := cli.Update(ctx, secret); err != nil {
 		return err
 	}
 
 	log.Info("User created", "user", user.UserID())
 	return nil
 }
+
+// getCustomUserSecret gets secret by name defined by `user.PasswordSecretRef.Name` or returns a secret
+// with newly generated password if name matches defaultName
+func getCustomUserSecret(ctx context.Context, cl client.Client, cr *api.PerconaServerMongoDB, name, defaultName, passKey string) (*corev1.Secret, error) {
+	log := logf.FromContext(ctx)
+
+	secret := &corev1.Secret{}
+	err := cl.Get(ctx, types.NamespacedName{Name: name, Namespace: cr.Namespace}, secret)
+
+	if err != nil && name != defaultName {
+		return nil, errors.Wrap(err, "failed to get user secret")
+	}
```

Comment:
I don't understand why we don't create the secret object if user just wants to customize its name. Was it like this in the spec? @spron-in @eleo007

Reply:
Not sure what do you mean by "user just wants to customize its name", but how this behaves is like this:
- If in the spec the user sets `passwordSecretRef.name` we will look for that secret. If we don't find it we will fail creating that user.
- If the user does not set `passwordSecretRef.name`, we will create secret `{cluster-name}-custom-user-secret`, generate a password for the user and set it by the key named after user name. 

And yes, I'll add this to the spec as well.